src/commands/admin/admin-subcommands/blacklist.ts

```diff
@@ -0,0 +1,216 @@
+/*
+ * MIT License
+ *
+ * Copyright (c) 2022-present Mirage Aegis
+ * 
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+
+import { 
+    SlashCommandSubcommandBuilder, EmbedBuilder, ChatInputCommandInteraction, User,
+    Collection, Guild, DiscordAPIError, ButtonBuilder, ButtonStyle, ActionRowBuilder,
+    Message, CollectorFilter, ButtonInteraction, MessageComponentInteraction
+} from "discord.js";
+import { Subcommand } from "../../../util/command-template.js";
+import { MISSING_PERMISSIONS, defaultErrorHandler } from "../../../util/error-handler.js";
+import { BlacklistT, Blacklist } from "../../../schemas/blacklist.js";
+import { createBlacklist } from "../../../util/create-blacklist.js";
+import { RED } from "../../../util/colours.js";
+
+/*
+ * A command for the administrators of Sushi Bot to use for
+ * blacklisting users
+ */
+
+// 7 days in seconds
+const BAN_DELETE_SECONDS: number = 604800;
+const name: string = "blacklist";
+
+export const command: Subcommand = {
```

Comment:
This is the `/admin blacklist` command for administrators to blacklist malicious users and ban them across all of Sushi Bot's servers